### PR TITLE
feat: CHK-2403 add metadata for application

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- add `metadata` for `Application` domain and document

#### Motivation and Context

The goal of this PR is to add `metadata` property for an `Application` .
This `metadata` is required to store a set of properties that characterize an `Application Wallet`, for example for a wallet enable to pagoPA payment (application PAGOPA), if the wallet was onboarded during a payment it can be represented as follows:
```
"applications": [{
   ....
  "metadata" : {
    "onboardeWithPayment" : "true"
 }
}]

```

So the idea is to model the metadata as a generic document, so in a domain as a HashMap (considering that a [Document](https://www.mongodb.com/docs/drivers/java/sync/v4.3/fundamentals/data-formats/documents/#overview) implements `Map<>`).

#### How Has This Been Tested?

- unit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/documents/wallets/Application.kt [hunk 1]
```diff
@@ -11,13 +12,15 @@ data class Application(
     val id: String,
     val name: String,
     val status: String,
-    val lastUpdateDate: String
+    val lastUpdateDate: String,
+    val metadata: Map<String, String>
```
Comment (Fernando-Granato) [inline src/main/kotlin/it/pagopa/wallet/documents/wallets/Application.kt:16]: Isn't it better to have **details** ?

**Review by Fernando-Granato — COMMENTED** [review]
Just a little comment

File: src/main/kotlin/it/pagopa/wallet/documents/wallets/Application.kt (hunk 1 above)
Comment (infantesimone) [reply src/main/kotlin/it/pagopa/wallet/documents/wallets/Application.kt:16]: good point, the idea is to have a general data related to a wallet. `details` implies strong typing, so fixed property and not a generic property with `Map`

Comment (Fernando-Granato) [reply src/main/kotlin/it/pagopa/wallet/documents/wallets/Application.kt:16]: Yes yes in fact I was only talking about the name of the attribute of the class.
```suggestion
    val details: Map<String, String>
    
```

**Review by Fernando-Granato — APPROVED** [review]

**Pull request merged**